### PR TITLE
Make generate-service-descriptor thread-safe again

### PR DESCRIPTION
**Pull request by wendigo**
SimpleDateFormat is not thread-safe which causes plugin to fail with the cryptic errors like `For input string: "E.404E2"`

### CODE REVIEW

File: src/main/java/io/trino/maven/ServiceDescriptorGenerator.java [hunk 1]
```diff
@@ -49,7 +49,7 @@ public class ServiceDescriptorGenerator
         extends AbstractMojo
 {
     private static final String LS = System.lineSeparator();
-    private static final DateFormat OUTPUT_DATE_FORMAT = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssXXX");
+    private static final ThreadLocal<DateFormat> OUTPUT_DATE_FORMAT = ThreadLocal.withInitial(() -> new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssXXX"));
```
Comment (findepi) [inline src/main/java/io/trino/maven/ServiceDescriptorGenerator.java:52]: here we could use `java.time.format.DateTimeFormatter`

**Review by findepi — APPROVED** [review]

File: src/main/java/io/trino/maven/ServiceDescriptorGenerator.java (hunk 1 above)
Comment (wendigo) [reply src/main/java/io/trino/maven/ServiceDescriptorGenerator.java]: PTAL now

**Comment by wendigo** [comment]
This is now on-par with the maven-assembly plugin outputTimestamp handling

**Comment by wendigo** [comment]
```
(base) ➜  target git:(master) ✗ ls -la io/trino/plugin/mysql
total 248
drwxr-xr-x  15 mateuszgajewski  staff    480 May 16 00:57 .
drwxr-xr-x   3 mateuszgajewski  staff     96 May 16 00:57 ..
-rw-r--r--   1 mateuszgajewski  staff    494 May 16 00:57 ImplementAvgBigint.class
-rw-r--r--   1 mateuszgajewski  staff   1430 May 16 00:57 MySqlClient$1.class
-rw-r--r--   1 mateuszgajewski  staff   1474 May 16 00:57 MySqlClient$2.class
-rw-r--r--   1 mateuszgajewski  staff   1437 May 16 00:57 MySqlClient$3.class
-rw-r--r--   1 mateuszgajewski  staff    872 May 16 00:57 MySqlClient$4.class
-rw-r--r--   1 mateuszgajewski  staff   5111 May 16 00:57 MySqlClient$ColumnHistogram.class
-rw-r--r--   1 mateuszgajewski  staff   1749 May 16 00:57 MySqlClient$ColumnIndexStatistics.class
-rw-r--r--   1 mateuszgajewski  staff   7264 May 16 00:57 MySqlClient$StatisticsDao.class
-rw-r--r--   1 mateuszgajewski  staff  64410 May 16 00:57 MySqlClient.class
-rw-r--r--   1 mateuszgajewski  staff   5391 May 16 00:57 MySqlClientModule.class
-rw-r--r--   1 mateuszgajewski  staff   2201 May 16 00:57 MySqlConfig.class
-rw-r--r--   1 mateuszgajewski  staff   1561 May 16 00:57 MySqlJdbcConfig.class
-rw-r--r--   1 mateuszgajewski  staff    446 May 16 00:57 MySqlPlugin.class
```

```
(base) ➜  target git:(master) ✗ ls -la META-INF/services
total 8
drwxr-xr-x  3 mateuszgajewski  staff   96 May 29 12:12 .
drwxr-xr-x  5 mateuszgajewski  staff  160 May 29 12:15 ..
-rw-r--r--  1 mateuszgajewski  staff   34 May 16 00:57 io.trino.spi.Plugin
```

**Pull request merged**